### PR TITLE
fix(editor): Ensure toasts show above modal overlays

### PR DESCRIPTION
**Pull request by mutdmour**
## Summary
Ensure notifications show above modal overlays
Screenshot 2024-10-25 at 11 54 09

## Review / Merge checklist

- [X] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/editor-ui/src/composables/useToast.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export interface NotificationErrorWithNodeAndDescription extends ApplicationErro
 const messageDefaults: Partial<Omit<NotificationOptions, 'message'>> = {
 	dangerouslyUseHTMLString: false,
 	position: 'bottom-right',
-	zIndex: 1900, // above NDV and below the modals
+	zIndex: 2100, // above NDV and modal overlays
```
Comment (netroy) [inline packages/editor-ui/src/composables/useToast.ts:23]: Can we maintain all various z-index values as `const enum` in a single place? That way it'll be easier to keep track of all the visual layers in app.

**Review by netroy — CHANGES_REQUESTED** [review]

File: packages/editor-ui/src/composables/useToast.ts (hunk 1 above)
Comment (mutdmour) [reply packages/editor-ui/src/composables/useToast.ts:23]: I thought about that but how would you handle ones set as css?

Comment (mutdmour) [reply packages/editor-ui/src/composables/useToast.ts:23]: we could keep a list of css variables or always try to use js

Comment (netroy) [reply packages/editor-ui/src/composables/useToast.ts:23]: **any** way that can help us keep track of all the major layers in a single place works for me. I'm indifferent about the actual implementation, as long as we can debug issues like this by looking in a single place in the codebase.

Comment (mutdmour) [reply packages/editor-ui/src/composables/useToast.ts:23]: makes sense. I think I can do it by generating css vars from an object.

File: packages/editor-ui/src/composables/useStyles.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+const APP_Z_INDEXES = {
```
Comment (netroy) [inline packages/editor-ui/src/composables/useStyles.ts:1]: This is so much better than random numbers spread across the codebase. Thanks for addressing this 🙇🏽

File: packages/editor-ui/src/composables/useStyles.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	DRAGGABLE: 9999999,
+	ACTIVE_STICKY: 9999999,
+	WORKLFOW_PREVIEW_NDV: 9999999,
+} as const;
+
+const setAppZIndexes = () => {
+	Object.keys(APP_Z_INDEXES).forEach((key) => {
+		const variableName = `--z-index-${key.toLowerCase().replaceAll('_', '-')}`;
+		const value = APP_Z_INDEXES[key as keyof typeof APP_Z_INDEXES];
+		document.documentElement.style.setProperty(variableName, `${value}`);
```
Comment (netroy) [inline packages/editor-ui/src/composables/useStyles.ts:24]: cool trick 😉

**Review by netroy — DISMISSED** [review]
👏🏽

File: packages/editor-ui/src/components/Sticky.vue [hunk 4]
```diff
@@ -54,6 +54,7 @@ const ndvStore = useNDVStore();
 const nodeTypesStore = useNodeTypesStore();
 const uiStore = useUIStore();
 const workflowsStore = useWorkflowsStore();
+const { APP_Z_INDEXES } = useStyles();
```
Comment (netroy) [inline packages/editor-ui/src/components/Sticky.vue:57]: ah, missing import here.

Comment (mutdmour) [reply packages/editor-ui/src/components/Sticky.vue:57]: thanks

**Comment by mutdmour** [comment]
@netroy can you please review again? Should be ready now 🤞🏽

**Review by netroy — APPROVED** [review]
👏🏽

**Comment by janober** [comment]
Got released with [`n8n@1.66.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.66.0)

**Pull request merged**